pkg/controller/directvolumemigration/rsync.go

```diff
@@ -124,11 +148,26 @@ func (t *Task) generateSSHKeys() error {
 		return err
 	}
 
-	t.SSHKeys = &sshKeys{
-		PublicKey:  &privateKey.PublicKey,
-		PrivateKey: privateKey,
+	publicRsaKey, err := ssh.NewPublicKey(&privateKey.PublicKey)
+	if err != nil {
+		return err
 	}
-	return nil
+
+	pubKeyBytes := ssh.MarshalAuthorizedKey(publicRsaKey)
+	secret := corev1.Secret{
+		ObjectMeta: metav1.ObjectMeta{
+			Namespace: migapi.OpenshiftMigrationNamespace,
+			Name:      "directvolumemigration-ssh-keys",
+			Labels: map[string]string{
+				"app": "directvolumemigration-rsync-transfer",
```

Comment:
+1 This label means our pre-existing deletion code will work in garbage collecting this secret after a successful migration.

Reply:
@alaypatel07 actually.... we won't clean this up with this PR. I took the approach of "create it if it doesn't exist, otherwise use it". So subsequent DVMs would use the same key.

If you think that's a bad idea we need to add openshift-migration NS to the cleanup code which is simple